src/main/java/core/basesyntax/NoSuchElementException.java

```diff
@@ -0,0 +1,7 @@
+package core.basesyntax;
+
+public class NoSuchElementException extends RuntimeException {
```

Comment:
It's good practice to provide a default serial version UID for a Serializable class. Since RuntimeException implements Serializable, you should add a generated serialVersionUID field.